phpcs.xml.dist

```diff
@@ -198,6 +222,20 @@
     ref="SlevomatCodingStandard.ControlStructures.RequireNullCoalesceOperator"
   />
 
+  <rule ref="SlevomatCodingStandard.TypeHints.PropertyTypeHint">
+      <exclude-pattern>scoper.inc.php</exclude-pattern>
+      <exclude-pattern>web-stories.php</exclude-pattern>
+      <exclude-pattern>includes/compat/*</exclude-pattern>
+      <exclude-pattern>tests/*</exclude-pattern>
+      <exclude-pattern>includes/templates/*</exclude-pattern>
+      <exclude-pattern>includes/templates/*</exclude-pattern>
+      <exclude name="SlevomatCodingStandard.TypeHints.PropertyTypeHint.UselessAnnotation" />
+  </rule>
+
+  <rule ref="SlevomatCodingStandard.TypeHints.PropertyTypeHint.MissingNativeTypeHint">
+    <exclude-pattern>includes/Services.php</exclude-pattern>
```

Comment:
Service class is used to class instance for plugin and services. The type hints can wrong, as service it does not always return instances of service, it can be a class that is not of type service.

Reply:
Hmm I think the issue you saw is that the fixer used union types like `private static Injector|Service $injector;`, which is a PHP 8+ thing.

To disable, we'll have to do

```xml
  <rule ref="SlevomatCodingStandard.TypeHints.PropertyTypeHint">
    <properties>
      <property name="enableMixedTypeHint" value="false" />
      <property name="enableUnionTypeHint" value="false" />
      <property name="enableIntersectionTypeHint" value="false" />
      <property name="enableStandaloneNullTrueFalseTypeHints" value="false" />
    </properties>
  </rule>
```

This is something we should do either way.

Then, another conflict in this file is that `SlevomatCodingStandard.TypeHints.PropertyTypeHint.UselessAnnotation` wants us to drop the useless `@var` annotations, but `Squiz.Commenting.VariableComment.MissingVar` wants us to add one nonetheless. So two conflicting rules.

I think we can safely disable the `Squiz.Commenting.VariableComment.MissingVar` rule

Reply:
https://github.com/GoogleForCreators/web-stories-wp/pull/12672#discussion_r1025334382